docs/v1/P4Runtime-Spec.mdk

```diff
@@ -4544,7 +4564,14 @@ four replicas of an ARP packet. These replicas will appear in the egress
 pipeline as independent packets with egress port set to PSA device port numbers
 corresponding to SDN port numbers 5, 12, 18 and 24. For more discussion on the
 translation between SDN ports and PSA device ports, refer to the
-[PSA Metadata Translation](#sec-translation-of-port-numbers) section.
+[PSA Metadata Translation](#sec-translation-of-port-numbers) section. Each
+replica can optionally have a list of backup replicas used as fallback ports
+for multicast. When the primary port (replica) goes down, the system
+uses the first backup replica for which its port is up. Whenever the primary
+port goes back up, the system will use the primary port again. In the case
+when the primary replica's ports and the backup replicas' ports are down,
+nothing is done for recovery until the primary replica or the backup replicas's
+ports go back up.
```

Comment:
Thanks for those added details.

There are still (at least) two possibilities for what you expect to happen when the primary and all backups are down for a particular replica, which due to packet processing side effects are different from each other.  Do you prefer one behavior over the other?

+ no replica made, and thus no side effects from after-multicast-replication processing should occur for this replica

vs

+ send one replica to the primary port, which will be dropped, but perform any side effects of after-multicast-replication processing

Reply:
A third option: Leave it as an implementation-specific behavior whether it does the first thing, or the second thing.  You have no preference.

Reply:
What do you think about option 4:

The behavior is the same it was prior to introducing fallback ports

Reply:
That's a good one!  I like it.

Reply:
Although it might raise the question in the minds of some readers: "What _was_ the behavior prior to introducing fallback ports" ?

Reply:
Specifying the behavior prior to how fallback ports was introduce might not be the best place.  I think that it might be up to the P4 architecture to define this behavior

Reply:
A probably-common implementation of this feature is one similar to the watch port feature.  For the watch port feature on action selectors, a typical implementation is that some driver software running locally on the switch monitors port up/down status, and edits the action selector contents in hardware as it detects ports going up and down, to remove any group elements that are associated with ports that are down.

For this feature, the similar implementation would be for driver software that is monitoring up/down status to edit multicast replication lists programmed into the hardware, as you specify here.  That is just one possible implementation, of course.

Such an implementation _could_ completely remove any entry for the replica in the hardware state when the primary and all backup ports are down for a particular replica.  It _could_ also leave some replica in the hardware state in this situation.

If you say you are willing to leave it up to the P4 architecture to specify this, then perhaps you do not have any preference between these choices?

Reply:
Yes that's right, I believe that we leave it as an implementation-specific behavior then, with no strong preference

Reply:
I think it might be nice in the PR to at least mention these two possible behaviors, and recommend that implementors use one of them.

Reply:
Updated the documentation to specify the two possible behaviors as suggested. Marking this as resolved.